### PR TITLE
adds payload encryption to task queueing [hybrid]

### PR DESCRIPTION
**Pull request by rdallman**
```
ironcli worker queue --cluster $hybrid_cluster --encryption-key=$encryption_key --payload "hello!" hello
```

entire process to be documented / described elsewhere (will update PR with said location)

also needed some small updates which I can bleed over to `iron_go3`
for a new field in the task when queueing to swapi.

### CODE REVIEW

**Comment by rdallman** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;wdc3fIbM&#x2F;5-encryption-for-payloads-and-maybe-env-vars-for-hybrid-workers"> Encryption for payloads (and maybe env vars?) for Hybrid Workers</a>

**Comment by rdallman** [comment]
@treeder should we have a warning on upload to make sure that cluster being queued into is a hybrid cluster that is configured with the decryption key properly ? or any other thoughts here ?

**Comment by rdallman** [comment]
still leaves schedules payloads as plain, can swoop in and fix this soon

File: worker.go [hunk 1]
```diff
@@ -118,3 +123,37 @@ func pushCodes(zipName string, w *worker.Worker, args worker.Code) (*worker.Code
...
+		return "", err
+	}
+
+	pbytes := []byte(payloadPlain)
+
+	// The IV needs to be unique, but not secure. Therefore it's common to
+	// include it at the beginning of the ciphertext.
+	ciphertext := make([]byte, aes.BlockSize+len(pbytes))
+	iv := ciphertext[:aes.BlockSize]
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
```
Comment (nikhilm) [inline worker.go]: Nit: `rand.Read(iv)` ?

**Comment by edsrzf** [comment]
Do we need authenticity as well as confidentiality? AES-CFB provides only confidentiality.

In other words, if I'm an attacker and I get access to the encrypted payloads, I can modify them and the workers will still try to use them. Of course, I have no idea what I'm modifying the payload to or from, but I can still potentially create havoc.

Here's a decent primer: http://crypto.stackexchange.com/questions/12178/why-should-i-use-authenticated-encryption-instead-of-just-encryption

(I have no context here so sorry if I'm bringing up stuff that's already been discussed or is irrelevant.)

**Comment by rdallman** [comment]
updated. thanks @edsrzf, could you review updates please

**Pull request merged**